metrics/api/src/main/java/io/helidon/metrics/api/Wrapped.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.metrics.api;
+
+/**
+ * Behavior of a type that wraps a related type.
+ */
+public interface Wrapped {
+
+    /**
+     * Unwraps the meter registry as the specified type.
+     *
+     * @param c {@link Class} to which to cast this meter registry
+     * @return the meter registry cast as the requested type
+     * @param <R> type to cast to
+     */
+    default <R> R unwrap(Class<? extends R> c) {
+        return c.cast(this);
```

Comment:
The implementation should be a bit more sofisticated - see `TracerResolverBuilder:128` as an example

Reply:
`TracerResolverBuilder#unwrap` seems like a slightly-rearranged copy of the implementation of `Class#cast(Object)`, except for
1. using an extra call of `this.getClass()` to pass to `builderClass#isAssignableFrom` (instead of just invoking `builderClass#isInstance(this)`), and
2. throwing an `IllegalArgumentException` instead of a `ClassCastException`. 

A developer using `unwrap` knows its purpose, so it throwing a `ClassCastException` would be natural.

So I guess I'm just not seeing the value our own code is adding beyond what the JDK's `Class#cast` already does.

Reply:
OK, in that case we maybe should change the implementation in that place to simple cast ;)